### ISSUE TITLE
Poor diagnostic wrong number of Type Arguments...

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#34620]
... with a type that exists in both a generic and non generic version

**Version Used**: 

master

**Steps to Reproduce**:

Compile the following code

```C#
using System;
public class C : IComparable<int, int> {}
```

https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEAMACFBGA3AWACgUBmLAJgwGEMQMBJKgezAAcIEJgAbOAHgCWAOxgAaDMJgA+DAG8AvkA

**Expected Behavior**:

error CS0305: Using the generic type 'IComparable<T>' requires 1 type arguments

**Actual Behavior**:

error CS0308: The non-generic type 'IComparable' cannot be used with type arguments

This error message is just plain false

**Comment by gafter** [comment]
The diagnostic is precisely correct.  See https://docs.microsoft.com/en-us/dotnet/api/system.icomparable?view=netframework-4.7.2

**Comment by sharwell** [comment]
I agree with @YairHalberstadt here. A similar situation occurs with `ValueTuple<int, int, int, int, int, int, int, int, int>`. Rather than assume the user meant to reference `ValueTuple`, it seems like the generic type with the closest number of type parameters should be used as the reference. For the original case above, this would be `IComparable<T>`. For the case I added, it would be `ValueTuple<T1, T2, T3, T4, T5, T6, T7, TRest>`.

**Comment by CyrusNajmabadi** [comment]
This was updated to now be:

<img width="1157" height="347" alt="Image" src="https://github.com/user-attachments/assets/6a840700-a493-4d28-889e-52081cff6b20" />
